src/plugins/VersionPlugin.ts

```diff
@@ -0,0 +1,93 @@
+import { AbstractPlugin, IMetaData } from '../plugin/AbstractPlugin'
+import PluginAPI from '../plugin/PluginAPI'
+import Translation from '@util/Translation';
+import Client from '../Client'
+
+const MIN_VERSION = '4.0.0';
+const MAX_VERSION = '99.0.0';
+
+const NAMESPACE_VERSION = 'jabber:iq:version'
+
+export default class VersionPlugin extends AbstractPlugin {
+
+   public static getId(): string {
+      return 'version';
+   }
+
+   public static getName(): string {
+      return 'Software Version';
+   }
+
+   public static getMetaData(): IMetaData {
+      return {
+         description: Translation.t('setting-version'),
+         xeps: [{
+            id: 'XEP-0092',
+            name: 'Software Version',
+            version: '1.1',
+         }]
+      }
+   }
+
+   constructor(pluginAPI: PluginAPI) {
+      super(MIN_VERSION, MAX_VERSION, pluginAPI);
+
+      let connection = pluginAPI.getConnection();
+
+      pluginAPI.addFeature(NAMESPACE_VERSION);
+
+      connection.registerHandler(this.onReceiveQuery, NAMESPACE_VERSION, 'iq');
+
+   }
+
+   //if name, version, os are empty it will be used as query for retrieve informations
+   public sendVersion(idstr: string, jid: string, name?: string, version?: string, os?: string): Promise<{}>
+   {
+      let iq = $iq({
+         type: (!name&&!version&&!os?'get':'result'),
+         to: jid,
+         id: idstr,
+         xmlns: 'jabber:client'
+      }).c('query', {
+         'xmlns': NAMESPACE_VERSION
+      });
+
+      if (name)
+      {
+          iq.c('name').t(name).up();
+      }
+
+      if (version)
+      {
+          iq.c('version').t(version).up();
+      }
+
+      if (os)
+      {
+          iq.c('os').t(os).up();
+      }
+
+      return this.pluginAPI.sendIQ(iq);
+   }
+
+   private onReceiveQuery = (stanza) => {
+        let OSName='Unknown OS';
```

Comment:
Would prefer to send an empty string, because this way other clients can correctly interpret the result.

Reply:
changed...